### PR TITLE
Nakamoto signer docs Updates

### PR DESCRIPTION
**Pull request by setbern**
## Description

Updated User guide section in regards towards running a sBTC Signer
- Added new page "Nakamoto Signer Get Started"
 - this includes the latest planned configuration set up for running a signer, text describing how to run a custom signer if needed
 - Added a page describing the lack of signer in Developer release

### CODE REVIEW

File: src/nakamoto-signer-developer-guide.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+# Disclaimer
+
+Development is on going and changes to the Signer configuration should be expected. This document will be updated every sprint to ensure its accuracy.
+
+## Prerequisites
+
+Rust - [To install](https://www.rust-lang.org/tools/install).
+
+Accessible Stacks node - A Stacks node running the stackerDB instance which is used for signer communication and transaction monitoring and broadcasting.
```
Comment (jferrant) [inline src/nakamoto-signer-developer-guide.md:15]: Might be good to link to stackerDB documentation somewhere? Even just linking to the Stackerdb description
https://github.com/stacks-network/stacks-blockchain/blob/develop/stackslib/src/net/stackerdb/mod.rs maybe?

File: src/nakamoto-signer-developer-guide.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+git clone git@github.com:stacks-network/stacks-blockchain.git
+```
+
+2. Next, navigate to the stacks-signer directory:
+
+```console
+cd stacks-blockchain/stacks-signer
+```
+
+3. Checkout the appropriate release branch you wish to use if you are not using the default main branch
```
Comment (jferrant) [inline src/nakamoto-signer-developer-guide.md:39]: I think stacks-network actually uses master, not main. So should update to that.

File: src/nakamoto-signer-developer-guide.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+1. First, download the precompiled default [TODO:NEED:LINK](LINK).
+
+2. Untar the file
+
+```console
+tar -xvf signer_binary.tar
+```
+
+3.  Check Extracted Files:
+    After running the untar command, the contents of the tar file should be extracted to the current directory. You should see the signer binary (stacks-signer-mini) and the configuration file (signer.toml) listed among the extracted files.
```
Comment (jferrant) [inline src/nakamoto-signer-developer-guide.md:63]: change stacks-signer-mini to stacks-signer

File: src/nakamoto-signer-developer-guide.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+
+## Prerequisites
+
+Rust - [To install](https://www.rust-lang.org/tools/install).
+
+Accessible Stacks node - A Stacks node running the stackerDB instance which is used for signer communication and transaction monitoring and broadcasting.
+
+Accessible Bitcoin node - A Bitcoin node used for transaction monitoring and broadcasting.
+
+A Stacks Address - Identify your address as a Signer
```
Comment (jferrant) [inline src/nakamoto-signer-developer-guide.md:19]: Might be more accurate to say "A Stacks Private Key - For signing sBTC transactions and validating blocks."

File: src/nakamoto-signer-developer-guide.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+| --------------------------- | -------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `signer_private_key`        | `true`   | Stacks private key of the signer, used for signing sBTC transactions.                                                                                        |
+| `stacks_node_rpc_url`       | `true`   | Stacks node RPC URL that points to a node running the stackerDB instance which is used for signer communication and transaction monitoring and broadcasting. |
+| `bitcoin_node_rpc_url`      | `true`   | Bitcoin node RPC URL used for transaction monitoring and broadcasting.                                                                                       |
+| `revealer_rpc_url`          | `true`   | Revealer RRC URL                                                                                                                                             |
```
Comment (jferrant) [inline src/nakamoto-signer-developer-guide.md:80]: Some additional Required config options (which of cousre may change...)
stackerdb_event_endpoint = "localhost:30000" (an endpoint for receiving events from the StackerDB contract)
stackerdb_contract_id = "ST11Z60137Y96MF89K1KKRTA3CR6B25WY1Y931668.signers-stackerdb" (the StackerDB qualified contract ID for Signer communication)

Comment (jferrant) [inline src/nakamoto-signer-developer-guide.md:80]: This revealer_rpc_url is going away. Can remove it.

File: src/how-to-signer.md [hunk 6]
```diff
@@ -1,53 +1,67 @@
 # How to operate a signer
-As a Stacker, you are responsible for ensuring the accuracy and integrity of incoming sBTC transactions. To satisfy this requirement and continue to earn BTC rewards, you must operate your own Signer or delegate your signing power to a third-party Signer. Below is a step-by-step guide to setting up and operating a Signer either by using the [default Signer implementation](#setting-up-a-default-signer) or by [utilizing the Signer SDK](#custom-signer-implementation).
 
-## Prerequisites
-Rust. To install, please [follow these instructions](https://www.rust-lang.org/tools/install).  
+As a Stacker, you are responsible for ensuring the accuracy and integrity of incoming sBTC transactions. To satisfy this requirement and continue to earn BTC rewards, you must operate your own Signer or delegate your signing power to a third-party Signer. Below is a step-by-step guide to setting up and operating a Signer either by using the [default Signer implementation](./nakamoto-signer-developer-guide.md) or by [utilizing the Signer SDK](./nakamoto-signer-developer-guide.md#custom-signer-implementation).
```
[the remark this answers is not in the document]
Comment (setbern) [reply src/how-to-signer.md]: Resolved in latest commit 97695ec9ce37137f9836f3e62bd22c7701fb51de

File: src/nakamoto-signer-developer-guide.md (hunk 1 above)
Comment (setbern) [reply src/nakamoto-signer-developer-guide.md:15]: resolved in 97695ec9ce37137f9836f3e62bd22c7701fb51de

File: src/nakamoto-signer-developer-guide.md (hunk 2 above)
Comment (setbern) [reply src/nakamoto-signer-developer-guide.md:39]: fixed 97695ec9ce37137f9836f3e62bd22c7701fb51de

File: src/nakamoto-signer-developer-guide.md (hunk 3 above)
Comment (setbern) [reply src/nakamoto-signer-developer-guide.md]: fixed 97695ec9ce37137f9836f3e62bd22c7701fb51de

File: src/nakamoto-signer-developer-guide.md (hunk 4 above)
Comment (setbern) [reply src/nakamoto-signer-developer-guide.md]: fixed 97695ec9ce37137f9836f3e62bd22c7701fb51de

File: src/nakamoto-signer-developer-guide.md (hunk 5 above)
Comment (setbern) [reply src/nakamoto-signer-developer-guide.md]: fixed 97695ec9ce37137f9836f3e62bd22c7701fb51de

Comment (setbern) [reply src/nakamoto-signer-developer-guide.md]: fixed 97695ec9ce37137f9836f3e62bd22c7701fb51de

**Review by jferrant — APPROVED** [review]

File: src/developer-release-signer.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+# Developer Release
+
+[The sBTC Developer Release](./sbtc-releases/sbtc-dev.md) is a minimum-viable testnet deployment of sBTC and was decided to be released without the without the complexity of adding a decentralized set of signers to validate sBTC deposits/withdrawals.
```
Comment (friedger) [inline src/developer-release-signer.md:3]: ~~the without~~

File: src/developer-release-signer.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
...
+
+Therefore our beloved the Developer Release aka Version 0.1 Alpha Romeo won't have signer when deployed to public test net.
```
Comment (friedger) [inline src/developer-release-signer.md:5]: Therefore our beloved Developer Release aka Version 0.1 Alpha Romeo only has one single signer that is operated by a central authority.

File: src/developer-release-signer.md (hunk 7 above)
Comment (setbern) [reply src/developer-release-signer.md]: fixed cbec4022241ad010c47e210512c1112d94883e68

File: src/developer-release-signer.md (hunk 8 above)
Comment (setbern) [reply src/developer-release-signer.md]: fixed cbec4022241ad010c47e210512c1112d94883e68

**Review by friedger — APPROVED** [review]
LGTM

**Pull request merged**